### PR TITLE
Randomize keeper fault injection settings in stress tests

### PR DESCRIPTION
**Pull request by devcrafter**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Follow up #42607 #43122

### CODE REVIEW

File: tests/clickhouse-test [hunk 1]
```diff
@@ -456,6 +456,8 @@ class SettingsRandomizer:
         "merge_tree_coarse_index_granularity": lambda: random.randint(2, 32),
         "optimize_distinct_in_order": lambda: random.randint(0, 1),
         "optimize_sorting_by_input_stream_properties": lambda: random.randint(0, 1),
+        "insert_keeper_max_retries": lambda: random.randint(20, 20),
+        "insert_keeper_fault_injection_probability": lambda: random.uniform(0, 0.01),
```
Comment (davenger) [inline tests/clickhouse-test:460]: This setting is already a probability of fault being injected. Does it make sense to randomize it?

Comment (devcrafter) [reply tests/clickhouse-test:460]: Agree, looks very probabilistically

**Review by davenger — APPROVED** [review]

**Comment by devcrafter** [comment]
AST fuzzer - created #43202

Integration test failures:
- `test_keeper_four_word_command/test.py::test_cmd_stat` - fix #43115
- `test_merge_tree_s3/test.py::test_table_manipulations[node]`- known #43169

**Comment by tavplubix** [comment]
There are two issues with this PR:
 - SettingsRandomizer in clickhouse-test is used for functional tests, so this PR enables fault injections in functional tests (seems like it was not expected). Yes, it randomizes settings in stress tests too, but as a side effect (because stress tests just run functional tests in many threads).
 - Random settings are used in BC check as well. So we have the same issue as we had in https://github.com/ClickHouse/ClickHouse/pull/42607#issuecomment-1312573472. All queries in BC check fail with `Code: 115. DB::Exception: Setting insert_keeper_max_retries is neither a builtin setting nor started with the prefix 'custom_' registered for user-defined settings. (UNKNOWN_SETTING)` (you can check `clickhouse-server.backward.stress.log`), so it tests nothing. Unfortunately, our scripts for stress tests cannot detect this automatically, so checks were green.

**Comment by davenger** [comment]
@tavplubix has very valid points, we have to revert this change :(

**Pull request merged**